### PR TITLE
Add loading MySQL/PG/OG view meta data impl

### PR DESCRIPTION
**Pull request by zjcnb**
For #20681

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java [hunk 1]
```diff
@@ -53,34 +58,64 @@ public Collection<String> getAllTableNames() {
         return tables.keySet();
     }
     
+    /**
+     * Get all view names.
+     *
+     * @return all view names
+     */
+    public Collection<String> getAllViewNames() {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java:66]: Does all view names are lowercase?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java [hunk 2]
```diff
@@ -53,34 +58,64 @@ public Collection<String> getAllTableNames() {
...
+     * @param view view
+     */
+    public void putView(final String viewName, final ShardingSphereView view) {
+        views.put(viewName.toLowerCase(), view);
+    }
+    
     /**
      * Add tables.
      *
+     *
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java:113]: Please remove this useless black line.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package org.apache.shardingsphere.infra.metadata.database.schema.loader.model;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+/**
+ * Table meta data.
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java:26]: Please rename `Table meta data` to `View meta data`.

File: shardingsphere-infra/shardingsphere-infra-context/src/main/java/org/apache/shardingsphere/infra/context/refresher/type/DropViewStatementSchemaRefresher.java [hunk 4]
```diff
@@ -39,8 +39,9 @@ public final class DropViewStatementSchemaRefresher implements MetaDataRefresher
     public Optional<MetaDataRefreshedEvent> refresh(final ShardingSphereDatabase database, final Collection<String> logicDataSourceNames,
                                                     final String schemaName, final DropViewStatement sqlStatement, final ConfigurationProperties props) throws SQLException {
         SchemaAlteredEvent event = new SchemaAlteredEvent(database.getName(), schemaName);
+        // TODO Drop view meta data from ShardingSphereSchema 's views
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-context/src/main/java/org/apache/shardingsphere/infra/context/refresher/type/DropViewStatementSchemaRefresher.java:42]: Do you think `ShardingSphereSchema's views` is better?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java (hunk 1 above)
Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/decorator/model/ShardingSphereSchema.java:66]: Yes, all view names are lowercase.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java (hunk 3 above)
Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java:26]: Done

File: shardingsphere-infra/shardingsphere-infra-context/src/main/java/org/apache/shardingsphere/infra/context/refresher/type/DropViewStatementSchemaRefresher.java (hunk 4 above)
Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-context/src/main/java/org/apache/shardingsphere/infra/context/refresher/type/DropViewStatementSchemaRefresher.java]: i will adjustment notes

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/SchemaMetaData.java [hunk 5]
```diff
@@ -25,8 +25,8 @@
 import java.util.Collection;
 
 /**
- * Schema meta data.
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/SchemaMetaData.java:28]: Why change the javadoc style?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.ToString;
+
+/**
+ * View meta data.
+ */
+@RequiredArgsConstructor
+@Getter
+@EqualsAndHashCode
+@ToString
+public class ViewMetaData {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/ViewMetaData.java:32]: Please add final for ViewMetaData.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/SchemaMetaData.java (hunk 5 above)
Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/model/SchemaMetaData.java:28]: No anything change in my local env. Let me check it

**Review by strongduanmu — APPROVED** [review]
LGTM, merged.

**Pull request merged**